### PR TITLE
Cleanup/Simplify Passcode & Biometric Unlock Flows

### PR DESCRIPTION
**Pull request by brandonpage**
- SFBiometricUnlockState enum to take the place of checking multiple booleans.
- Stop round-tripping data from SFSecurityLockout to nav/view controllers that never utilize the data.
- Stop passing passcode to view/nav controllers that don't need it.
- Cleanup integration examples

WIP because tests will be added.

### CODE REVIEW

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFSecurityLockout.h [hunk 1]
```diff
@@ -72,23 +72,31 @@ typedef NS_ENUM(NSUInteger, SFSecurityLockoutAction) {
...
-typedef struct {
-    NSUInteger passcodeLength;
-    NSUInteger lockoutTime;
-    BOOL biometricUnlockAllowed;
-} SFAppLockConfigurationData;
+typedef NS_ENUM(NSUInteger, SFBiometricUnlockState) {
+    /**
+     Touch Id or Face Id should be the default unlock method.
+     **/
+    SFBiometricUnlockUserAllowed,
```
Comment (wmathurin) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFSecurityLockout.h]: Should we call it `SFBiometricUnlockUserApproved`?

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFSecurityLockout.h [hunk 2]
```diff
@@ -72,23 +72,31 @@ typedef NS_ENUM(NSUInteger, SFSecurityLockoutAction) {
...
+    
+    /**
+     User declined biometric unlock, do not prompt again.
+     **/
+    SFBiometricUnlockUserDeclined,
+    
+    /**
+     User has not been prompted to use biometric unlock.
+     **/
+    SFBiometricUnlockPromptUser,
```
Comment (wmathurin) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFSecurityLockout.h]: Should we call it `SFBiometricUnlockAvailable` ?

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/AppLockView/SFSDKAppLockViewController.m [hunk 3]
```diff
@@ -95,60 +91,61 @@ - (void)setupNavBar
...
+        [self.navigationController popViewControllerAnimated:NO];
+        [SFSecurityLockout unlock:YES action:SFSecurityLockoutActionPasscodeVerified];
     }
 }
 
 - (void)passcodeFailed
 {
     [[SFPasscodeManager sharedManager] resetPasscode];
-    [SFSecurityLockout unlock:NO action:SFSecurityLockoutActionNone passcodeConfig:self.configData];
+    [SFSecurityLockout unlock:NO action:SFSecurityLockoutActionNone];
```
Comment (wmathurin) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/AppLockView/SFSDKAppLockViewController.m]: Should we have two methods:
- one called when unlock succeeded: `+ (void)unlock:(SFSecurityLockoutAction)action`
- one called when unlock failed (should logout users etc): `+ (void) clearAppState`
Then we would no longer need the SFSecurityLockoutActionNone in the enum.

**Comment by brandonpage** [comment]
Found an issue with the notification approach.

**Comment by wmathurin** [comment]
Finishing up that PR here: https://github.com/forcedotcom/SalesforceMobileSDK-iOS/pull/2687

**Pull request closed without merging**